itou/employee_record/management/commands/process_asp_report_file.py

```diff
@@ -0,0 +1,98 @@
+import json
+import logging
+import os.path as os_path
+
+from django.core.management.base import BaseCommand
+from rest_framework.renderers import JSONRenderer
+
+from itou.employee_record.models import EmployeeRecord, EmployeeRecordBatch
+from itou.employee_record.serializers import EmployeeRecordSerializer
+
+
+class Command(BaseCommand):
+    """
+    Manually process an employee record ASP report file
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        handler = logging.StreamHandler(self.stdout)
+
+        self.logger = logging.getLogger(__name__)
+        self.logger.propagate = False
+        self.logger.addHandler(handler)
+
+        self.logger.setLevel(logging.INFO)
+
+    def add_arguments(self, parser):
+        """
+        Command line arguments
+        """
+        parser.add_argument("--file", dest="input_file", type=open, required=True, help="ASP input file")
+
+    def handle(self, *args, **options):
+        """
+        Fixes employee record "in-between" state in case of the crash of CRON jobs
+        """
+        input_file = options.get("input_file")
+        renderer = JSONRenderer()
+
+        with open(input_file.name, "r") as f:
+            data = json.load(f)
+            filename = os_path.basename(f.name)
+            batch_filename = EmployeeRecordBatch.batch_filename_from_feedback(filename)
+            cnt = 0
+            asp_success_code = "0000"
+
+            self.logger.info("Start processing of ASP file: %s", batch_filename)
+
+            records = data.get("lignesTelechargement")
+
+            for _, employee_record in enumerate(records, 1):
```

Comment:
je ne saisis pas la différence avec `for employee_record in records:`, y en a-t-il une ?